demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/ApplicationHomeController.scala

```diff
@@ -1,34 +1,132 @@
 
   val logger = Logger("ApplicationHomeController")
 
+  val loginForm = Form(
+    mapping(
+      "userName" -> nonEmptyText(minLength = 5, maxLength = 10),
+      "password" -> nonEmptyText
+    )(LoginCredentials.apply)(LoginCredentials.unapply)
+  )
+
   /**
     * Render index page
+    *
     * @return result to view
     */
-  def indexPage() : Action[AnyContent] = Action.async { implicit req =>
-    logger info("ApplicationHome: Rendering Index page")
-    Future.successful(
+
+  def home(): Action[AnyContent] = Action { implicit req =>
+
+    req.session.get("api-key").map { apiKey =>
+      logger info ("ApplicationHome: Rendering Index page")
       Ok(uk.gov.ons.addressIndex.demoui.views.html.index(version))
-    )
+    }.getOrElse {
+      Redirect(uk.gov.ons.addressIndex.demoui.controllers.routes.ApplicationHomeController.login())
+    }
+
+  }
+
+  /**
+    * Load login viewlet unless config says login is not required in the conf
+    * @return
+    */
+  def login: Action[AnyContent] = Action { implicit req =>
+    logger.info("loginRequired " + conf.config.loginRequired )
+    if (!conf.config.loginRequired)
+    {
+      Redirect(uk.gov.ons.addressIndex.demoui.controllers.routes.ApplicationHomeController.home())
+        .withSession("api-key" -> "")
+    }
+    else {
+      Ok(uk.gov.ons.addressIndex.demoui.views.html.login("", version))
+    }
+  }
+
+  /**
+    * Redirect to the API gateway to perform login
+    * @return
+    */
+  def doLogin: Action[AnyContent] = Action { implicit req =>
+
+    val formValidationResult = loginForm.bindFromRequest.data
+    (for {
+      userName <- formValidationResult.get("userName") if userName.nonEmpty
+      password <- formValidationResult.get("password") if userName.nonEmpty
+    } yield {
+    //  use fake gateway in dev or test
+      val mode: Mode = environment.mode
+ //     val realGateway: Boolean = mode match {
+  //      case Mode.Dev => false
+  //      case Mode.Test => false
+  //      case Mode.Prod => true
+  //    }
+
+      val realGateway: Boolean = false
+
+      val fullURL = conf.config.gatewayURL+"/ai/login"
+  //    logger.info("full request = " + fullURL )
+      if (realGateway) {
+
+        val request: WSRequest = ws.url(conf.config.gatewayURL+"/ai/login")
+
+    //    logger.info("about to run request")
+        val complexRequest: WSRequest =
+          request.withHeaders("Accept" -> "application/json")
+            .withAuth(userName, password, WSAuthScheme.BASIC)
+            .withRequestTimeout(10000.millis)
+
+        val futureResponse: Future[WSResponse] = complexRequest.get()
+
+        val result = Await.result(futureResponse, 10000.millis)
+     //   logger.info("request run + result.status")
+
+        if (result.status != OK) {
```

Comment:
How does this work? If the correct result.status with token is not 200, than what is it?